templates/rpm/project.spec.erb

```diff
@@ -175,6 +228,23 @@ done
 <%- end -%>
 
 %preun
+# Run pre-uninstall scripts on upgrade if defined
+<%- scripts = get_preremove_actions("upgrade") -%>
```

Comment:
What if get_remove_actions returned ':' if it were empty? then this wouldn't need to care if scripts were empty, it could always add scripts.